packages/fuels-programs/src/lib.rs

```diff
@@ -21,3 +32,323 @@ impl Configurables {
         }
     }
 }
+
+pub struct ReceiptParser {
+    receipts: Vec<Receipt>,
+}
+
+impl ReceiptParser {
+    pub fn new(receipts: &[Receipt]) -> Self {
+        let relevant_receipts: Vec<Receipt> = receipts
+            .iter()
+            .filter(|receipt| {
+                matches!(receipt, Receipt::ReturnData { .. } | Receipt::Return { .. })
+            })
+            .cloned()
+            .collect();
+
+        Self {
+            receipts: relevant_receipts,
+        }
+    }
+
+    /// Based on receipts returned by a script transaction, the contract ID (in the case of a contract call),
+    /// and the output param, parse the values and return them as Token.
+    pub fn try_parse_output(
```

Comment:
I'd call it just `parse` since it is probably going to look like this at the call site:
```rust
receipts_parser.parse(...)?;
```